app/gosns/publish.go

```diff
@@ -74,8 +75,7 @@ func PublishV1(req *http.Request) (int, interfaces.AbstractResponseBody) {
 	return http.StatusOK, respStruct
 }
 
-func publishSQS(subscription *app.Subscription, topicName string, requestBody *models.PublishRequest) error {
-	messageAttributes := utils.ConvertToOldMessageAttributeValueStructure(requestBody.MessageAttributes)
+func publishSQS(subscription *app.Subscription, message string, messageAttributes map[string]app.MessageAttributeValue, subject string, topicName string, messageStructure string) error {
```

Comment:
I just undid all this argument fatigue in favor of referring to the `models.PublishRequest` request body.  If you're changing it back because we don't have a `PublishRequest` in the batch flow, I understand, but let's come up with something else to hold all this.  Perhaps an interface or something.  Having a bazillion arguments makes this difficult to juggle in my opinion.

If you really want to get slick, I wouldn't mind a helper method like `publishMessage` that takes a topic name, and all the message(s) data, then publishes them however they need to go.  It could route between the SQS and HTTP code and return successes VS failures.  It would reduce a lot of this duplicate complexity, but I leave that choice to you.